src/ints/bios.cpp

```diff
@@ -1211,14 +1212,25 @@ class BIOS final : public Module_base{
 		/* INT 12 Memory Size default at 640 kb */
 		callback[2].Install(&INT12_Handler,CB_IRET,"Int 12 Memory");
 		callback[2].Set_RealVec(0x12);
-		if (IS_TANDY_ARCH) {
+		if (machine == MCH_TANDY) {
 			/* reduce reported memory size for the Tandy (32k graphics memory
 			   at the end of the conventional 640k) */
-			if (machine==MCH_TANDY) mem_writew(BIOS_MEMORY_SIZE,624);
-			else mem_writew(BIOS_MEMORY_SIZE,640);
-			mem_writew(BIOS_TRUE_MEMORY_SIZE,640);
-		} else mem_writew(BIOS_MEMORY_SIZE,640);
-		
+			mem_writew(BIOS_MEMORY_SIZE, 624);
+			mem_writew(BIOS_TRUE_MEMORY_SIZE, 640);
+		} else if (machine == MCH_PCJR) {
+			const Section_prop* section = static_cast<Section_prop*>(control->GetSection("dosbox"));
+			assert(section);
+			if (section->Get_bool("pcjr_expanded_memory")) {
+				mem_writew(BIOS_MEMORY_SIZE, 640);
+				mem_writew(BIOS_TRUE_MEMORY_SIZE, 640);
+			} else {
+				mem_writew(BIOS_MEMORY_SIZE, 96);
```

Comment:
Would it make sense to introduce a `dos_memory.h` file and reference these memory size constants in `bios.cpp` and `dos_memory.cpp` instead of hardcoding and duplicating them?

Example for a few such constants (there are more):

### dos_memory.h

```cpp
constexpr auto PcjrVideoMemorySizeKb    = 32;
constexpr auto PcjrStandardMemorySizeKb = 128;
```

### dos_memory.cpp

```cpp
constexpr auto video_memory_start = kilobytes_to_segment(PcjrStandardMemorySizeKb - PcjrVideoMemorySizeKb); // instead of hardcoding 96
```

### bios.cpp

```cpp
mem_writew(BIOS_MEMORY_SIZE, PcjrStandardMemorySizeKb - PcjrVideoMemorySizeKb); // instead of hardcoding 96
mem_writew(BIOS_TRUE_MEMORY_SIZE, PcjrStandardMemorySizeKb); // instead of hardcoding 128
```

Reply:
Added the header with the constants like you suggested.